src/js/game/LevelMVC/LevelView.js

```diff
@@ -714,16 +714,29 @@ module.exports = class LevelView {
 
   playTrack(position, facing, isOnBlock, entity = this.player, completionHandler) {
     entity.onTracks = true;
-    const track = this.controller.levelModel.actionPlane.getMinecartTrack(position, facing);
 
-    if (!track) {
+    // Need to get track on current position to avoid mishandling immediate turns
+    let track = this.controller.levelModel.actionPlane.getMinecartTrack(position, facing);
+
+    let offset = FacingDirection.directionToOffset(facing);
+    let nextPos = [entity.position[0] + offset[0], entity.position[1] + offset[1]];
```

Comment:
this can use `Position.add`, or if we're not using `offset` elsewhere it can just be `let nextPos = Position.forward(entity.position, facing)`

Reply:
@Hamms Gotta love these new Position methods ❤️